libs/hmpb/src/marking.ts

```diff
@@ -0,0 +1,199 @@
+import PdfDocument from 'pdfkit';
+
+import { assert } from '@votingworks/basics';
+import {
+  ballotPaperDimensions,
+  Election,
+  GridPosition,
+  Rect,
+  Vote,
+  VotesDict,
+} from '@votingworks/types';
+
+import {
+  BUBBLE_HEIGHT_PX,
+  BUBBLE_WIDTH_PX,
+  pageMarginsInches,
+  TIMING_MARK_DIMENSIONS,
+} from './ballot_components';
+
+// NOTE: All values used in this module are in PDF user space `pt` units.
+
+const PT = 1;
+const IN = 72 * PT;
+const PX = IN / 96;
+
+const pageMargins = [
+  pageMarginsInches.left * IN,
+  pageMarginsInches.top * IN,
+] as const;
+
+const timingMarkCount = [34, 41] as const;
+const timingMarkSize = [
+  TIMING_MARK_DIMENSIONS.width * IN,
+  TIMING_MARK_DIMENSIONS.height * IN,
+] as const;
+
+const bubbleSize = [BUBBLE_WIDTH_PX * PX, BUBBLE_HEIGHT_PX * PX] as const;
+// We grow the mark slightly to help make up for slight variations in printing:
+const markSize = [bubbleSize[0] * 1.2, bubbleSize[1] * 1.2] as const;
+const markBorderRadius = markSize[1] * 0.5;
+const markSizeHalf = [markSize[0] * 0.5, markSize[1] * 0.5] as const;
+
+const writeInFontName = 'Roboto-Bold';
+const writeInFontSizeDefault = 12;
+const writeInFontSizeReduced = 10;
+
+/**
+ * Center of the top-left timing mark.
+ *
+ * [TODO] We may need to move this into the election definition, to support
+ * marking 3rd part ballots
```

Comment:
nit: "party"

Also, is this something we'll need to manually calibrate for each supported 3rd party ballot?